### PR TITLE
dns: consider network local endpoints for stateful sets

### PR DESCRIPTION
**Pull request by ramaraochavali**
Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/name_table.go [hunk 1]
```diff
@@ -78,8 +78,15 @@ func (configgen *ConfigGeneratorImpl) BuildNameTable(node *model.Proxy, push *mo
 			if svc.Attributes.ServiceRegistry == string(serviceregistry.Kubernetes) &&
 				svc.Resolution == model.Passthrough && len(svc.Ports) > 0 {
 				for _, instance := range push.ServiceInstancesByPort(svc, svc.Ports[0].Port, nil) {
-					// Add individual addresses even for cross cluster.
-					if instance.Endpoint.SubDomain != "" && instance.Endpoint.Network == node.Metadata.Network {
+					if instance.Endpoint.Network != node.Metadata.Network {
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/name_table.go:81]: I think we have some semantic like "if network is unset it should access all networks" that we may need to preserve?

**Comment by hzxuzhonghu** [comment]
I would suggest adding a comment saying we will remove the restriction after cross-network headless service can be accessed

File: pilot/pkg/networking/core/v1alpha3/name_table.go (hunk 1 above)
Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/name_table.go:81]: That is a great point. I will add that

**Comment by ramaraochavali** [comment]
@howardjohn added that check.
@hzxuzhonghu updated the comment

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**